Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Upsert.ts

```diff
@@ -20,6 +20,31 @@ export async function upsertDocument(
 
   try {
     await session.withTransaction(async () => {
+      // Check whether this is an insert or update
+      const isInsert: boolean = (await mongoCollection.findOne({ _id: id }, onlyReturnId(session))) == null;
+
+      // If inserting a subclass, check whether the superclass identity is already claimed by a different subclass
+      if (isInsert && documentInfo.superclassInfo != null) {
+        const superclassAliasId: string = documentIdForSuperclassInfo(documentInfo.superclassInfo);
+        const superclassAliasIdInUse: boolean =
+          (await mongoCollection.findOne({ aliasIds: superclassAliasId }, onlyReturnId(session))) != null;
+
+        if (superclassAliasIdInUse) {
+          Logger.debug(
+            `mongodb.repository.Upsert.upsertDocument: Upserting document id ${id} failed due another subclass with the same identity`,
+            traceId,
+          );
+
+          upsertResult = {
+            response: 'INSERT_FAILURE_CONFLICT',
+            failureMessage: `Insert failed: the identity is in use by another resource with the same superclass`,
```

Comment:
Here's how the ODS/API responds when you try to create an LEA with an existing SchoolId:

```
{
  "message": "A natural key conflict occurred when attempting to create a new resource 'EducationOrganization' with a duplicate key.  The duplicated columns and values are [EducationOrganizationId] (5) This is likely caused by multiple resources with the same key in the same file. Exactly one of these resources was inserted."
}
```

How difficult would it be to detect what the superclass is (e.g. EducationOrganization) and include that in the message, instead of the word "superclass"? I don't think we need to perfectly match up the error message.